pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -555,14 +555,27 @@ def is_subrepo(directory):
                 namespace=directory,
             )
 
-            dv = RpmDeclarativeVersion(first_stage=stage, repository=repo, mirror=mirror)
-            repo_version = dv.create() or repo.latest_version()
-
-            repo_config["sync_details"]["most_recent_version"] = repo_version.number
-            repo.last_sync_details = repo_config["sync_details"]
-            repo.save()
-
-            repo_sync_results[directory] = repo_version
+            # Concurrent syncs colliding while attempting to create the same version in the same
+            # repo is Bad. It can happen if multiple repositories have the same sub-repositories
+            # (e.g., multiple repos syncing from the same remote kickstart-repo with sub-repos,
+            # such as RHEL6-kickstart).
```

Comment:
Add some details here about why the conflict happens - because of the way the naming scheme works

Reply:
:+1: